app/[locale]/dashboard/config/widget-registry.tsx

```
@@ -123,6 +125,38 @@ const PropfirmCatalogueWidget = dynamic(
+
+// Lightweight chart preview skeleton — no dynamic imports, pure CSS
+function ChartPreviewSkeleton({ title }: { title: string }) {
+  return (
+    <Card data-chart-surface="modern" className="h-[300px]">
+      <CardHeader className="pb-3">
+        <CardTitle className="text-sm font-medium">{title}</CardTitle>
+      </CardHeader>
+      <CardContent className="pb-2 flex flex-col gap-3">
+        <div className="flex items-end gap-1 h-24">
+          {Array.from({ length: 8 }, (_, i) => (
+            <div key={i} className="flex-1 rounded-t bg-background/25-foreground/10" style={{ height: `${30 + Math.random() * 70}%` }} />
```

ESLint

[error] Error: Cannot call impure function during render

`Math.random` is an impure function. Calling an impure function can produce unstable results that update unpredictably when the component happens to re-render. (https://react.dev/reference/rules/components-and-hooks-must-be-pure#components-and-hooks-must-be-idempotent).

/home/jailuser/git/app/[locale]/dashboard/config/widget-registry.tsx:139:111
  137 |         <div className="flex items-end gap-1 h-24">
  138 |           {Array.from({ length: 8 }, (_, i) => (
> 139 |             <div key={i} className="flex-1 rounded-t bg-background/25-foreground/10" style={{ height: `${30 + Math.random() * 70}%` }} />
      |                                                                                                               ^^^^^^^^^^^^^ Cannot call impure function
  140 |           ))}
  141 |         </div>
  142 |         <div className="h-3 w-3/4 bg-background/25-foreground/10 rounded" />

(react-hooks/purity)